lib/src/components/Checkbox/index.tsx

```diff
@@ -0,0 +1,30 @@
+import * as Ariakit from '@ariakit/react'
+import React, { forwardRef } from 'react'
+
+import { classNames } from '@/utils'
+
+import checkboxStyles from './checkbox.module.scss'
+import type { CheckboxProps } from './types'
+
+const cx = classNames(checkboxStyles)
+
+export const Checkbox = forwardRef<HTMLInputElement, CheckboxProps>(
+  ({ checked = false, indeterminate = false, name, onChange, variant, ...rest }, ref) => {
+    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+      e.target.checked = !e.target.checked
+      if (onChange) onChange(e)
+    }
+
+    return (
+      <Ariakit.Checkbox
+        checked={checked}
+        className={cx('root', variant && `variant-${variant}`, indeterminate && 'indeterminate')}
```

Comment:
The `className` is in the `rest` so it will override the whole style if provided
```suggestion
        className={cx('root', variant && `variant-${variant}`, indeterminate && 'indeterminate', className)}
```